File: typescript/base-hmi-types/src/screens/widgets/HmiToggleSwitch.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiColor, HmiProperty } from "../base.js";
import { HmiButtonBase } from "./HmiButtonBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";
import { HmiSwitchType } from "./HmiSwitchType.js";
import { HmiToggleSwitchShapeType } from "./HmiToggleSwitchShapeType.js";

export class HmiToggleSwitch extends HmiButtonBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiToggleSwitch;
  }

  mode?: HmiProperty<HmiSwitchType>;
  header?: HmiProperty<boolean>;
  headerText?: HmiProperty<HmiMultilingualText>;
  remark?: HmiProperty<HmiMultilingualText>;
  shapeType?: HmiProperty<HmiToggleSwitchShapeType>;
  offThumbColor?: HmiProperty<HmiColor>;
  onThumbColor?: HmiProperty<HmiColor>;
}

File: src/BaseHmiTypes/Screens/Widgets/HmiToggleSwitch.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Widgets;

public class HmiToggleSwitch : HmiButtonBase
{
    public HmiToggleSwitch()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiToggleSwitch;
    }

    public HmiProperty<HmiSwitchType>? Mode { get; set; }

    public HmiProperty<bool>? Header { get; set; }

    public HmiProperty<HmiMultilingualText>? HeaderText { get; set; }

    public HmiProperty<HmiMultilingualText>? Remark { get; set; }
}
